### PR TITLE
Loosened requirement for psr/http-message -  ^1.0|^2.0

### PR DESCRIPTION
**Pull request by Rastusik**
This PR was created according to #106

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   ],
   "require": {
     "php": ">=8.0",
-    "psr/http-message": "~1.0"
+    "psr/http-message": "^1.1|^2.0"
```
Comment (mad-briller) [inline composer.json:21]: this will increase the minimum as well as the maximum, this should probably be `^1.0` to ensure full backwards compatibility

Comment (Rastusik) [reply composer.json]: done

**Comment by Rastusik** [comment]
@mad-briller not sure what is wrong with the CI but I suppose it is not related to this PR, is it?

**Comment by ackintosh** [comment]
Yes, that's right. Some tests are flaky so I did rerun the CI and they're passed. 👌

**Comment by ackintosh** [comment]
I noticed that `psr/http-message` v2 is not used in the CI, will dig into that later.

https://github.com/ackintosh/ganesha/actions/runs/5901311195/job/16038538654?pr=107#step:5:53
> Locking psr/http-message (1.1)

**Comment by ackintosh** [comment]
I have updated guzzle (in require-dev) and confirmed that:

- psr/http-message 2.0 is used in `Tests (8.2)`
- psr/http-message 1.0 is used in `Tests (8.2, --prefer-lowest --prefer-stable)`
- All tests have passed 👍

**Comment by Rastusik** [comment]
Thank s for the effort :)

**Review by ackintosh — APPROVED** [review]

**Comment by ackintosh** [comment]
This PR has been released as v3.1.2. 🎉  https://github.com/ackintosh/ganesha/releases/tag/3.1.2

@Rastusik Thanks for your contribution! @mad-briller Thanks for your review!

**Pull request merged**